### PR TITLE
Do not format certain attributes of State on abnormal epgsql sock termination

### PR DESCRIPTION
**Pull request by enidgjoleka**
Imagine the following scenario:
We have fired an `epgsql:equery/squery` call and while getting the results back from PG server we are accumulating rows in the `epgsql_sock` state. While this is happening `epgsql_sock` process terminates abnormally. 
At this moment gen_server will try to print/format the State.
Given that we might have accumulated several hundred thousands or even million of rows in the state this will cause a memory spike and in the worst case scenario an OOM.

This solution tries to solve that problem by introducing another gen_server optional callback, `format_status`, where we replace the accumulated list in the rows with an atom stating `information_redacted`.

While looking at the code I realized that `format_status` could be used to remove the sensitive info  from `connect_opts` that we are currently hiding via lambdas.

### CODE REVIEW

**Comment by seriyps** [comment]
Thanks!
Regarding password masking, I'm afraid it still has a chance to show up in crash logs as "last message in was ...", so, I'd prefer to leave it as is =)
Otherwise, looks good and makes a lot of sense

**Comment by enidgjoleka** [comment]
You are right, password shows up as last message. 
I dropped the second commit for now :+1:

**Review by seriyps — APPROVED** [review]
Let's see if anyone else have smth to say, otherwise will merge in a couple of days.

File: src/epgsql_sock.erl [hunk 1]
```diff
@@ -298,6 +298,13 @@ terminate(_Reason, #state{mod = ssl, sock = Sock}) -> ssl:close(Sock).
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
+format_status(normal, [_PDict, State=#state{}]) ->
+  [{data, [{"State", State}]}];
+format_status(terminate, [_PDict, State]) ->
+  %% Do not format the rows attribute when process terminates abnormally
+  %% but allow it when is a sys:get_status/1.2
+  State#state{rows = information_redacted}.
```
Comment (davidw) [inline src/epgsql_sock.erl:306]: In the shell, there's some kind of thing to print a term... but only up to a certain size. I forget what it is, but maybe something like that could be used to print a bit, but truncate it?

Comment (enidgjoleka) [reply src/epgsql_sock.erl:306]: We introduced a similar tuning in our system that limits the size up to 50-100 but we did not have this in place for many many years until the last incident.
Hence, let's try to be defensive and not assume all the systems that use `epgsql` have this kind of tuning in place.

**Review by anha0825 — APPROVED** [review]

**Comment by seriyps** [comment]
Sorry, forgot about it. Merged, thanks!

**Pull request merged**